contracts/bondingcurve/CommonsToken.sol

```diff
@@ -221,10 +221,12 @@ contract CommonsToken is BondingCurveToken, Pausable {
     uint256 paidExternal = initialContributions[msg.sender].paidExternal;
     uint256 lockedInternal = initialContributions[msg.sender].lockedInternal;
 
+    uint256 hatchInternal = paidExternal * p0;
+
     // The total amount of INTERNAL tokens that should have been unlocked.
-    uint256 shouldHaveUnlockedInternal = (paidExternal * unlockedInternal) / initialRaise;
+    uint256 shouldHaveUnlockedInternal = hatchInternal * (lockedHatchInternal / hatchLimitInternal);
```

Comment:
You have flipped the logic of `unlockedInternal` to `lockedHackedInternal` and as far as I can understand in this version the proportion is going in opposite direction , before it was tending to 1 and not it tends to 0. Also I am concerned that this operation leads to float numbers as you are always using internal units, and before as we were dividing internal by external to avoid that

Reply:
The reasoning for flipping the logic of `unlockedInternal` to `lockedHatchedInternal` is because it is important for the application layer to know how many locked tokens are remaining. Also, it simplified calculations...

However, I agree this is adding a new feature and should be a separate PR if we are going to add this functionality.

I have created a new PR that fixes only the two bugs.

https://github.com/lightstreams-network/genesis-contracts/pull/5